ui/desktop/app/routes/scopes/scope/projects/sessions/index.js

```diff
@@ -39,53 +34,43 @@ export default class ScopesScopeProjectsSessionsIndexRoute extends Route {
    *        are now filtered on the client by projects and status,
    *        while user_id filtering remains server side.
    *
-   * @return {Promise{[SessionModel]}}
+   * @return {Promise<{sessions: [SessionModel], allSessions: [SessionModel]}>}
    */
-  async model() {
-    const { id: scope_id } = this.modelFor('scopes.scope');
-    const queryOptions = {
-      scope_id,
-      recursive: true,
-      include_terminated: this.status?.includes('terminated'),
-    };
+  async model({ targets }, transition) {
+    const from = transition.from?.name;
 
-    // Recursively query sessions within the current scope for the current user
-    let sessions = await this.resourceFilterStore.queryBy(
-      'session',
-      { user_id: this.session.data.authenticated.user_id },
-      queryOptions,
-    );
+    const filters = {
+      user_id: { equals: this.session.data.authenticated.user_id },
+      target_id: [],
+    };
+    targets.forEach((target) => {
+      filters.target_id.push({ equals: target });
+    });
 
-    // If project filters are selected, filter sessions by the selected projects
-    if (this.project?.length) {
-      const projectIDs = this.project.map((p) => p?.id);
-      sessions = sessions.filter((s) => projectIDs.includes(s?.scopeID));
-    }
+    const queryOptions = { query: { filters }, force_refresh: true };
+    const sessions = await this.store.query('session', queryOptions);
 
-    // If status filters are selected, filter sessions by the selected statuses
-    if (this.status?.length) {
-      sessions = sessions.filter((s) => this.status.includes(s?.status));
+    // Query all sessions and all targets for defining filtering values if entering route for the first time
+    if (from !== 'scopes.scope.projects.sessions.index') {
+      const options = { pushToStore: false };
+      this.allSessions = await this.store.query(
```

Comment:
What is `allSessions` used for?

Reply:
it is used in the controller file index.js line 37 to make sure that all target filter options are available for existing sessions.